src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexMatchTimeoutException.cs

```diff
@@ -66,61 +54,22 @@ public RegexMatchTimeoutException(string message)
         /// </summary>
         /// <param name="message">The error message that explains the reason for the exception.</param>
         /// <param name="inner">The exception that is the cause of the current exception, or a <code>null</code>.</param>
-        public RegexMatchTimeoutException(string message, Exception inner)
-            : base(message, inner)
-        {
-            Init();
-        }
-
-        private void Init()
-        {
-            Init("", "", TimeSpan.FromTicks(-1));
-        }
-
-        private void Init(string input, string pattern, TimeSpan timeout)
-        {
-            _regexInput = input;
-            _regexPattern = pattern;
-            _matchTimeout = timeout;
-        }
-
-        public string Pattern
-        {
-            [SecurityCritical]
-            get
-            { return _regexPattern; }
-        }
-
-        public string Input
-        {
-            [SecurityCritical]
-            get
-            { return _regexInput; }
-        }
-
-
-        public TimeSpan MatchTimeout
-        {
-            [SecurityCritical]
-            get
-            { return _matchTimeout; }
-        }
+        public RegexMatchTimeoutException(string message, Exception inner) : base(message, inner) { }
 
         protected RegexMatchTimeoutException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            string input = info.GetString("regexInput");
-            string pattern = info.GetString("regexPattern");
-            TimeSpan timeout = new TimeSpan(info.GetInt64("timeoutTicks"));
-            Init(input, pattern, timeout);
+            Input = info.GetString("regexInput");
+            Pattern = info.GetString("regexPattern");
+            MatchTimeout = new TimeSpan(info.GetInt64("timeoutTicks"));
         }
 
-        void ISerializable.GetObjectData(SerializationInfo si, StreamingContext context)
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
         {
-            base.GetObjectData(si, context);
-            si.AddValue("regexInput", _regexInput);
-            si.AddValue("regexPattern", _regexPattern);
-            si.AddValue("timeoutTicks", _matchTimeout.Ticks);
+            base.GetObjectData(info, context);
+            info.AddValue("regexInput", Input);
+            info.AddValue("regexPattern", Pattern);
+            info.AddValue("timeoutTicks", MatchTimeout.Ticks);
```

Comment:
Since you're modifying this code, can you verify we have a test for serialization of a RegexMatchTimeoutException, and if we don't, add one?  There's a helper in Common you can use that'll serialize and deserialize an exception, verify properties before and after, etc.